src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,85 @@
+package duke.task;
+
+import duke.Parser;
+import duke.Ui;
+import duke.task.Task;
+
+import java.util.ArrayList;
+public class TaskList {
+    private ArrayList<Task> tasks;
+    protected static Ui ui = new Ui();
+    protected int size = 0;
+    protected static int max = 100;
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+        this.size = tasks.size();
+    }
+
+    public TaskList() {
+        this.tasks = new ArrayList<Task>(max);
+    }
+
+    private Task getTask(int index) {
+        return this.tasks.get(index);
+    }
+
+    /**
+     * Sets the taskList to point to a new taskArray
+     * @param taskArrayList
+     */
+    public void setTasks(ArrayList<Task> taskArrayList) {
+        this.tasks = taskArrayList;
+        this.size = taskArrayList.size();
+    }
+
+    public int getSize() {
+        return this.size;
+    }
+
+    public void addTask(Task task) {
+        this.tasks.add(task);
+        this.size++;
+    }
+
+    public void deleteTask(int index) {
+        this.tasks.remove(index);
+        this.size--;
+    }
+
+    public String getSearchTask(String searchString) {
+        String result = "";
+        Parser parser = new Parser();
+        for (Task task : tasks) {
+            if (parser.parseSearch(task.getDescription(), searchString)) {
+                result += task + "\n";
+            }
+        }
+        return result;
+    }
+
+    public void setTaskDone(int index, boolean done) {
```

Comment:
I like how you named the methods clearly as verbs. They are easy to understand :+1: